source/common/network/listener_filter_buffer_impl.cc

```diff
@@ -11,10 +11,10 @@ ListenerFilterBufferImpl::ListenerFilterBufferImpl(IoHandle& io_handle,
                                                    ListenerFilterBufferOnDataCb on_data_cb,
                                                    uint64_t buffer_size)
     : io_handle_(io_handle), dispatcher_(dispatcher), on_close_cb_(close_cb),
-      on_data_cb_(on_data_cb), buffer_(std::make_unique<uint8_t[]>(buffer_size)),
-      base_(buffer_.get()), buffer_size_(buffer_size) {
+      on_data_cb_(on_data_cb), zero_buffer_size_(buffer_size == 0),
```

Comment:
We also can add a parameter called 'on_data_callback_disabled' in the constructor of ListenerFilterbuffer. And rename `zero_buffer_size_` to `on_data_callback_disabled_`.

Reply:
Updated it to on_data_callback_disabled_